packages/loader/src/gltf/GLTFUtils.ts

```diff
@@ -134,135 +134,91 @@ export class GLTFUtils {
+        }
+        return bufferInfo;
+      });
+    } else {
+      const count = accessorCount * dataElementSize;
+      const data = new TypedArray(count);
+      const bufferInfo = new BufferInfo(data, false, elementStride);
+      bufferInfo.restoreInfo = new BufferDataRestoreInfo(
+        new RestoreDataAccessor(undefined, TypedArray, undefined, count)
+      );
       if (accessor.sparse) {
         return GLTFUtils.processingSparseData(context, accessor, bufferInfo).then(() => bufferInfo);
       }
-      return bufferInfo;
-    });
-  }
-
-  public static bufferToVector3Array(
-    data: TypedArray,
-    byteStride: number,
-    accessorByteOffset: number,
-    count: number
-  ): Vector3[] {
-    const bytesPerElement = data.BYTES_PER_ELEMENT;
-    const offset = (accessorByteOffset % byteStride) / bytesPerElement;
-    const stride = byteStride / bytesPerElement;
-
-    const vector3s = new Array<Vector3>(count);
-    for (let i = 0; i < count; i++) {
-      const index = offset + i * stride;
-      vector3s[i] = new Vector3(data[index], data[index + 1], data[index + 2]);
+      return Promise.resolve(bufferInfo);
     }
-    return vector3s;
   }
 
-  /**
-   * @deprecated
-   * Get accessor data.
-   */
-  static getAccessorData(glTF: IGLTF, accessor: IAccessor, buffers: ArrayBuffer[]): TypedArray {
-    const bufferViews = glTF.bufferViews;
-    const bufferView = bufferViews[accessor.bufferView ?? 0];
-    const arrayBuffer = buffers[bufferView.buffer];
-    const accessorByteOffset = accessor.hasOwnProperty("byteOffset") ? accessor.byteOffset : 0;
-    const bufferViewByteOffset = bufferView.hasOwnProperty("byteOffset") ? bufferView.byteOffset : 0;
-    const byteOffset = accessorByteOffset + bufferViewByteOffset;
-    const accessorTypeSize = GLTFUtils.getAccessorTypeSize(accessor.type);
-    const length = accessorTypeSize * accessor.count;
-    const byteStride = bufferView.byteStride ?? 0;
-    const arrayType = GLTFUtils.getComponentType(accessor.componentType);
-    let uint8Array;
-    if (byteStride) {
-      const accessorByteSize = accessorTypeSize * arrayType.BYTES_PER_ELEMENT;
-      uint8Array = new Uint8Array(accessor.count * accessorByteSize);
-      const originalBufferView = new Uint8Array(arrayBuffer, bufferViewByteOffset, bufferView.byteLength);
-      for (let i = 0; i < accessor.count; i++) {
-        for (let j = 0; j < accessorByteSize; j++) {
-          uint8Array[i * accessorByteSize + j] = originalBufferView[i * byteStride + accessorByteOffset + j];
-        }
+  static bufferToVector3Array(
+    buffer: TypedArray,
+    byteOffset: number,
+    count: number,
+    normalized: boolean,
+    componentType: AccessorComponentType
+  ): Vector3[] {
+    const baseOffset = byteOffset / buffer.BYTES_PER_ELEMENT;
+    const stride = buffer.length / count;
+    const vertices = new Array<Vector3>(count);
+    if (normalized) {
+      const factor = GLTFUtils.getNormalizedComponentScale(componentType);
```

Comment:
Here can simplify as:
```
const factor = normalized ? GLTFUtils.getNormalizedComponentScale(componentType) : 1; 
```

Reply:
> Here can simplify as:
> 
> ```ts
> const factor = normalized ? GLTFUtils.getNormalizedComponentScale(componentType) : 1; 
> for (let i = 0; i < count; i++) {
>   const index = baseOffset + i * stride;
>   vertices[i] = new Vector3(buffer[index] * factor, buffer[index + 1] * factor, buffer[index + 2] * factor);
> }
> ```

Although it is a bit of a waste of performance, it is a one-time load, OK!

Reply:
done